nilearn/maskers/nifti_labels_masker.py

```diff
@@ -22,12 +22,18 @@ def __init__(self, _resampled_labels_img_, background_label, strategy,
         self.mask_img = mask_img
 
     def __call__(self, imgs):
-        from ..regions import signal_extraction
+        from ..regions.signal_extraction import (
+            _img_to_signals_labels_with_masked_atlas,
+        )
 
-        return signal_extraction.img_to_signals_labels(
-            imgs, self._resampled_labels_img_,
-            background_label=self.background_label, strategy=self.strategy,
-            keep_masked_labels=self.keep_masked_labels, mask_img=self.mask_img)
+        signals, labels, masked_labels_img =\
+            _img_to_signals_labels_with_masked_atlas(
```

Comment:
So now we can go back to using img_to_signal_labels and setting the new parameter to True to return the masked atlas right?

Reply:
You mean I should remove `_img_to_signals_labels_with_masked_atlas` function and replace it with `img_to_signals_labels` with `return_masked_atlas=True`?

Reply:
Well since we decided to go with deprecating the return behavior of `img_to_signals_labels` I'm not sure we even need `_img_to_signals_labels_with_masked_atlas` anymore since it was originally written to avoid deprecation. I would rather get rid of it and revert `img_to_signals_labels` to what it was except for the added deprecation code. And then yes, call `img_to_signals_labels` with `return_masked_atlas=True`